controller/engine_controller.go

```diff
@@ -717,9 +729,10 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 		}
 
 		engine.Status.LastRestoredBackup = engine.Spec.RequestedBackupRestore
+		logrus.Infof("updating engine %v", engine.Name)
 		engine, err = m.ds.UpdateEngine(engine)
 		if err != nil {
-			logrus.Errorf("failed to do update engine %v after incremental restoration: %v", engine.Name, err)
+			logrus.Errorf("failed to do update engine after incremental restoration: %v", err)
```

Comment:
Revert back to the old way.

Reply:
This was a bug in the old code as this part of the code will be hit when err is not nil and engine is nil.
`engine.Name` results in nil pointer dereferencing. Do you want me to revert back?

Reply:
We can assign the engineName before the UpdateEngine() call then.